### PR TITLE
Add `grad_norm` metrics

### PR DESCRIPTION
**Pull request by yzhangcs**
### What does this PR do?

This PR adds the `grad_norm` metric to the logging metrics.

### Why this PR is important

Monitoring the gradient norm is crucial for detecting training stability and getting some optimization insights.

### CODE REVIEW

File: torchtitan/components/metrics.py [hunk 1]
```diff
@@ -394,7 +397,7 @@ def log(
         color = self.color
         logger.info(
             f"{color.red}step: {step:2}  "
-            f"{color.green}loss: {global_avg_loss:7.4f}  "
+            f"{color.green}loss: {global_avg_loss:7.4f} gnorm: {grad_norm:.2f} "
```
Comment (tianyu-l) [inline torchtitan/components/metrics.py:400]: image

I agree this metric would be helpful. Do you think it has to be in the terminal print, instead of staying in TB/W&B only for now?
If so, could you invent a new color in the `Color` class -- o/w it seems visually harder to recognize loss. Also nit: two spaces before after each column.

**Review by tianyu-l — COMMENTED** [review]
had a comment
also pls fix linting
